### ISSUE TITLE
Alist Format Problem

### ISSUE THREAD

**Issue by page0egap** [daniestevez/ldpc-toolbox#2]
According to https://rptu.de/en/channel-codes/matrix-file-formats, the alist format is consistent with irregular ldpc codes. For variable node, if the degree is less than the max degree, padding with zeroes is needed. However, in `sparse` module, i find the function `from_alist` does not deal with this situation. The error `attempt to subtract with overflow` would be thrown when meeting these padding zeroes. 

I'm not sure whether this is a flaw or feature. But I think it might be helpful to implement `from_alist` for irregular ldpc.

**Comment by daniestevez** [comment]
Good point. I had overlooked this sentence in [MacKays's website](http://www.inference.org.uk/mackay/codes/alist.html) when implementing alists:

> NB: If the rows or columns are irregular, you must pad the low-weight rows/columns with zeroes so as to make the two sets of lists regular.

I have used alists produced by ldpc-toolbox with other tools, such as AFF3CT toolbox, but I don't remember if the codes I used where regular by chance.

Definitely this is an easy fix. When reading with `from_alist` both the correct format with zeros and the wrong format I implemented without zeros can be handled. The reader can simply ignore the zeros if they're present. For writing an alist I should implement the correct format with zeros, providing the wrong format under a different function in case anyone finds it useful (and marking this as deprecated, perhaps).
